pennylane/interfaces/execution.py

```diff
@@ -587,6 +587,20 @@ def gradient_fn(internal_tapes):
                 pass_kwargs=True,
                 return_tuple=False,
             )
+
+            # Adjoint Jacobian with backward pass and jitting needs the original circuit output state which
+            # can not be reused from the device if `grad_on_execution is False`.
+
+            interface_jax = interface
+            if interface == "jax":
+                from .jax import get_jax_interface_name
+
+                interface_jax = get_jax_interface_name(tapes)
+            if interface_jax == "jax-jit":
+                use_device_state = gradient_kwargs.get("use_device_state", None)
+                if use_device_state:

```

Comment:
do users set this somewhere? if they do, we should probably raise a warning saying that we're changing it

Reply:
I think it is not necessary because you expect the state to go from initial to final (execution) and then the state goes backward for adjoint. Here with jitting the execution step is not effective (it does not go from initial to final) so we have to mention it. At the end the results are the same and the simulation as well.